### ISSUE TITLE
Add support for more XML and JSON parsing libraries

### ISSUE THREAD

**Issue by metavida** [ruby-oembed/ruby-oembed#4]
The ruby-oembed gem currently relies on the xml-simple and json gems. It should allow for other parsing libraries to be either detected or specified. Libraries we should support:
### XML
- Hpricot
- Nokogiri
### JSON
- Rails 3 JSON

Sugest even more in the comments.

**Comment by hmans** [comment]
Hey @metavida, how about [Oj](https://github.com/ohler55/oj)? (I'm happy to have a stab at adding this.)

I'm using ruby-oembed in a non-Rails project right now, and it's failing with many JSON responses (esp. those coming from Twitter, but I haven't seen a clear pattern) unless I specifically require `active_support/json`, which I'm not _terribly_ keen on doing.

**Comment by metavida** [comment]
Hi @hmans,

Have you tried using the json gem instead of ActiveSupport's JSON? Either `require "json"` before you require ruby-oembed *or* call `OEmbed::Formatter::JSON.backend = 'JSONGem'`

I likely won't have time to implement support for Oj, but would be glad to review & merge in a pull request if you send one! You can [use this class as inspiration](https://github.com/ruby-oembed/ruby-oembed/blob/master/lib/oembed/formatter/json/backends/jsongem.rb). Really all you'll need is three public class methods! You can even start using your custom JSON formatter before I've officially merged it:
```
class MyOjFormatter
  ...
end

OEmbed::Formatter::JSON.backend = MyOjFormatter
```

Hope this helps!
--Marcos

**Comment by hmans** [comment]
Thanks for the heads-up!

I tried loading the `json` gem before loading `ruby-oembed` and got mixed results unfortunately -- I haven't had much luck in tracking down why this was, but it _did_ work at some point, and then stopped. It's good to know that I can set the backend explicitly, I will try that next. Otherwise, maybe I'll find some time to cram Oj in there soon.

Thanks, and keep up the good work!

**Comment by metavida** [comment]
Browsing through these very old issues and it looks like I **have** resolved this one! Version 0.8.0 of ruby-oembed added support for custom Formatters that can use any parsing library/technique that users want! https://github.com/ruby-oembed/ruby-oembed?tab=readme-ov-file#formatters
